### PR TITLE
Fix/fix mutex deadlock in worker

### PR DESCRIPTION
**Pull request by guilhermocc**
## What ❓ 
Use defer to unlock mutexes locks in OperationCancelFunctions and executors, refactor create_new_scheduler_version e2e tests.

## Why 🤔 
We have discovered an edge case in which the worker was tucked with a deadlock after canceling an active operation.

## How
- Fixing mutexes deadlock
- Refactor create_new_scheduler_version e2e tests

### CODE REVIEW

**Review by caiordjesus — APPROVED** [review]

File: internal/core/services/operation_manager/operation_cancel_functions_test.go [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+			args: args{
+				schedulerName: "scheduler1",
+				operationID:   "operation1",
+				cancelFunc:    func() {},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			of := &OperationCancelFunctions{
```
[the remark this answers is not in the document]
Comment (guilhermocc) [reply internal/core/services/operation_manager/operation_cancel_functions_test.go]: Nice suggestion!
